storage/innobase/row/row0umod.cc

```diff
@@ -1144,7 +1144,8 @@ row_undo_mod_upd_exist_sec(
 			/* This is only legitimate when
 			rolling back an incomplete transaction
 			after crash recovery. */
-			ut_a(thr_get_trx(thr)->is_recovered);
+			ut_a(thr_get_trx(thr)->is_recovered
+			     || thr_get_trx(thr)->in_rollback);
```

Comment:
Doesn’t `trx_t::in_rollback` always hold here? There should be an assertion about it in a code path that leads to here. It seems that the assertion needs to be removed altogether and the comment revised.